spec/unit/inventory_spec.rb

```diff
@@ -51,4 +51,52 @@
       expect(resource.children_by_type('type-03', true).size).to eq(1)
     end
   end
+
+  describe '#ResultFetcher' do
+    it 'should fetch two first pages' do
+      batches = [[1, 2], [3, 4], [5]]
```

Comment:
Those variables should be extracted to `let` statements.

Reply:
Can you detail what you expect?
If I understand let [1], it's meant to be defined on outer-scope for lazy evaluation. In this case lazy evaluation wouldn't bring anything as they are constants (except `calls_count`).
However I can move them to outer scope for less duplicates.

For `calls_count`, I tried to use let: (`let(:calls_count) { 0 }`), but it doesn't like the increment operation: `NoMethodError:
       undefined method '+' for nil:NilClass`.
Btw I think it's good to set it in test scope instead of outer scope, so that the test is self-explanatory.

[1] https://relishapp.com/rspec/rspec-core/v/3-4/docs/helper-methods/let-and-let